### ISSUE TITLE
Template property override

### ISSUE THREAD

**Issue by phlafoo** [mattyhall/rs-tiled#310]
I have an ellipse object that triggers some functionality when the player enters the zone defined by the ellipse shape. I have some custom properties for this object so I have converted it into a template to reuse in many places.

The issue I'm having is that when I try to resize the ellipse, the new width and height is not in the `ObjectData` anywhere. It only gives me the width and height that is defined in the template file. Perhaps I am missing something but I couldn't find anything in the docs about this.

Here is the object from my .tmx file:
```xml
<object id="8" template="../templates/Trigger.tx" type="Trigger" x="617" y="5565" width="100" height="100">
```

And here is the template:
```xml
<template>
 <object type="Trigger" width="111" height="289">
<!-- custom properties -->
  <ellipse/>
 </object>
</template>
```

In this case the `ObjectData` only contains a width of 111 and height of 289 but I would expect it to have a width and height of 100. Is this the intended behaviour? I have another similar template with the same issue but it is a rectangle.

I am new to Tiled so let me know if there is a better way to go about this.

**Comment by phlafoo** [comment]
Looking at the code [here](https://github.com/mapeditor/rs-tiled/blob/5864f8f9c2a62e08a905499675df49aac6de9509/src/objects.rs#L325) I think I found the problem.
```rust
// Possibly copy properties from the template into the object
// Any that already exist in the object's map don't get copied over
if let Some(templ) = template {
    shape.get_or_insert(templ.object.shape.clone());

    for (k, v) in &templ.object.properties {
        if !properties.contains_key(k) {
            properties.insert(k.clone(), v.clone());
        }
    }
}
```

When I am using the template, the object in the level file does not have the ellipse tag so the shape is not parsed at all. When this code is reached, `shape` is `None` and so the template shape is used. So it seems that the correct width and height can only be used _after_ getting the shape from the template since the type of shape cannot be known from the object xml alone.

**Comment by aleokdev** [comment]
This looks incorrect, but I'll wait for @bjorn's response

**Comment by bjorn** [comment]
Hmm, yes this is a bug. It is correct to inherit the shape from the template, but the size can be overridden by the object. This is also broken for point shapes, where the position stored in the shape currently ends up being the one of the template. I'm working on a fix.

**Comment by bjorn** [comment]
@phlafoo The issue should be fixed by #311, though I didn't do a test yet.
